### PR TITLE
Show case tile templates with maps only when CASE_TILE_MAP is enabled

### PR DESCRIPTION
**Pull request by MartinRiese**
## Product Description

Only show case list tile templates using maps when FF CASE_LIST_MAP is enabled.

## Technical Summary

https://dimagi-dev.atlassian.net/browse/QA-5322

## Feature Flag

CASE_LIST_TILE,  CASE_LIST_MAP

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by AddisonDunn — COMMENTED** [review]
I think it might be safer to just store the templates known to have a map in a list, and then check whether they're in that list.

I.e. in case_tiles.py:

`templates_with_map = [CaseTileTemplates.ONE_TWO_ONE_ONE, CaseTileTemplates.ONE_TWO_ONE, CaseTileTemplates.ONE_ONE_TWO]`

And then in the view:

`if template in templates_with_map`...

Or if you can put the list directly into the choices class (not sure of the specifics of `model.TextChoices`), that would work too.

**Comment by orangejenny** [comment]
Agree with @AddisonDunn , having a magic string in the template description is fragile. What about a flag in the template json file?

**Comment by MartinRiese** [comment]
@orangejenny Do you think using the fact that a field named "map" exists in the config is ok? Otherwise, I will add something like `has_map` to the config.

File: corehq/apps/app_manager/suite_xml/features/case_tiles.py [hunk 1]
```diff
@@ -32,6 +32,7 @@ class CaseTileTemplates(models.TextChoices):
 class CaseTileTemplateConfig:
     slug: str = ''
     filename: str = ''
+    has_map: str = ''
```
Comment (orangejenny) [inline corehq/apps/app_manager/suite_xml/features/case_tiles.py:35]: isn't this a boolean?

Comment (MartinRiese) [reply corehq/apps/app_manager/suite_xml/features/case_tiles.py:35]: fail :(

**Review by orangejenny — APPROVED** [review]

File: corehq/apps/app_manager/views/modules.py [hunk 2]
```diff
@@ -203,6 +203,13 @@ def _get_shared_module_view_context(request, app, module, case_property_builder,
     '''
     item_lists = item_lists_by_app(app, module) if app.enable_search_prompt_appearance else []
     case_types = set(module.search_config.additional_case_types) | {module.case_type}
+    case_list_map_enabled = toggles.CASE_LIST_MAP.enabled(app.domain)
+    case_tile_template_option_to_configs = \
```
[the remark this answers is not in the document]
Comment (MartinRiese) [reply corehq/apps/app_manager/views/modules.py:207]: I like that. much cleaner.

**Review by orangejenny — APPROVED** [review]

**Review by AddisonDunn — APPROVED** [review]

File: corehq/apps/app_manager/views/modules.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (MartinRiese) [reply corehq/apps/app_manager/views/modules.py]: @AddisonDunn Hers actually does not compile.

**Pull request merged**